src/java.base/share/classes/java/lang/String.java

```diff
@@ -2270,18 +2261,15 @@ public boolean startsWith(String prefix, int toffset) {
         if (toffset < 0 || toffset > length() - prefix.length()) {
             return false;
         }
-        byte ta[] = value;
-        byte pa[] = prefix.value;
+        byte[] ta = value;
+        byte[] pa = prefix.value;
         int po = 0;
         int pc = pa.length;
         byte coder = coder();
         if (coder == prefix.coder()) {
-            int to = (coder == LATIN1) ? toffset : toffset << 1;
-            while (po < pc) {
-                if (ta[to++] != pa[po++]) {
-                    return false;
-                }
-            }
+            toffset <<= coder;
+            return ArraysSupport.mismatch(ta, toffset,
+                    pa, 0, pc) < 0;
```

Comment:
`offset <<= coder` is only obvious if the reader knows the value of LATIN1, maybe it would be simpler to read if you kept "int to"?